pkg/collector/corechecks/containers/generic/check.go

```diff
@@ -14,6 +14,9 @@ import (
 	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
 	"github.com/DataDog/datadog-agent/pkg/util/containers"
 	"github.com/DataDog/datadog-agent/pkg/util/containers/v2/metrics"
+
+	// Register all the metrics collectors
+	_ "github.com/DataDog/datadog-agent/pkg/util/containers/v2/metrics/collectors"
```

Comment:
That's the main reason why the collectors were not in a dedicated package, I am not a big fan of this pattern, it caused multiple issues with the previous container provider design.

Reply:
I managed to get rid of that by restructuring a little bit the package structure and introducing [this package facade](https://github.com/DataDog/datadog-agent/pull/9612/files#diff-ac4d9a82ecab221f3444c208275d304d1c841bcd122555890cb71900ea7169f0).